### ISSUE TITLE
Invalid ClosedownAt setting

### ISSUE THREAD

**Issue by giulyko00** [IbcAlpha/IBC#248]
Hi and thanks for your precious work.

I'd like to know why do I get the error:

2024-01-28 22:50:54:487 IBC: Invalid ClosedownAt setting: 'Friday 23:30'; format should be: <[day ]hh:mm> eg 22:00 or Friday 22:00

What's wrong in my time format?

**Comment by giulyko00** [comment]
Ok thank you so much. This in fact was the issue. I thought English was the go-to language for anyone using latin characters. Changed to Italian and works. Thanks again!

**Comment by rlktradewright** [comment]
Yes, I realise this is slightly confusing. The Java API that IBC calls to validate the value is locale-dependent, which means it expects the value to be formatted according to the rules of the locale that the operating system is running in. So if your OS is running in Italian, it will expect Italian day-of-week names. One day I will get round to improving this to allow English to be used.

This is completely independent from the language that TWS is running in, which must be English for IBC: this is because IBC has to recognise dozens of different bits of text to be able to detect the various windows and controls it's interested in: they are displayed in the language that TWS is running in. It would be technically possible for IBC to do this detection regardless of the TWS language, because all these strings are held in a locales resource file that can be used to map back to a language-independent identifier which IBC could work with. This was suggested to me many years ago by the author of TwsStart, a Windows program with similar functions to IBC. But life is too short...
